src/widget/wknobcomposed.cpp

```diff
@@ -116,6 +114,14 @@ void WKnobComposed::onConnectedControlChanged(double dParameter, double dValue)
     }
 }
 
+void WKnobComposed::setDefaultAngleFromParameterOrReset(std::optional<double> parameter) {
+    if (!parameter.has_value() || parameter.value() < 0 || parameter.value() > 1) {
+        m_defaultAngle = std::nullopt;
+    } else {
+        m_defaultAngle = std::lerp(m_dMinAngle, m_dMaxAngle, parameter.value());
+    }
```

Comment:
not a suggestion, just a preview of what would be possible in C++23. (imo easier to read and also faster because it only does a single has_value check. 
```suggestion
    m_defaultAngle = parameter.transform([this](double value) {
        return (parameter < 0 || parameter > 1) ? 
            std::nullopt : 
            std::lerp(m_dMinAngle, m_dMaxAngle, parameter);
    });
```

Reply:
> In general I would prefer to give m_defaultAngle a valid value here. It is better to have a variable that can be used unconditional in all cases.

That defeats the purpose of the optional. This snippet of code here can't accurately determine what value is appropriate.